nacos-spring-context/src/main/java/com/alibaba/nacos/spring/context/annotation/EnableNacosAotProcessor.java

```diff
@@ -0,0 +1,93 @@
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package com.alibaba.nacos.spring.context.annotation;
+
+import com.alibaba.nacos.spring.util.aot.AotDetector;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanDefinitionRegistryPostProcessor;
+import org.springframework.beans.factory.support.DefaultListableBeanFactory;
+import org.springframework.context.EnvironmentAware;
+import org.springframework.core.annotation.AnnotationAttributes;
+import org.springframework.core.env.Environment;
+import org.springframework.core.type.AnnotationMetadata;
+
+import java.util.Map;
+
+import static com.alibaba.nacos.spring.util.NacosBeanUtils.*;
+
+/**
+ * {@link EnableNacos} AotProcessor
+ * Except for the operation of registering BeanDefinition, all other operations in {@link NacosBeanDefinitionRegistrar} must be done here
+ * because spring will not call {@link NacosBeanDefinitionRegistrar#registerBeanDefinitions} in AOT.
+ * @author SuperZ1999
+ */
+public class EnableNacosAotProcessor implements BeanDefinitionRegistryPostProcessor, EnvironmentAware, BeanFactoryAware {
+    private Environment environment;
+
+    private BeanFactory beanFactory;
+
+    @Override
+    public void postProcessBeanDefinitionRegistry(BeanDefinitionRegistry registry) throws BeansException {
+        if (!AotDetector.useGeneratedArtifacts()) {
+            return;
+        }
+        DefaultListableBeanFactory beanFactory = (DefaultListableBeanFactory) this.beanFactory;
+        Map<String, Object> beansWithAnnotation = beanFactory.getBeansWithAnnotation(EnableNacos.class);
+        for (Object bean : beansWithAnnotation.values()) {
+            Class<?> aClass = bean.getClass();
+            if (aClass.getAnnotation(EnableNacos.class) == null) {
+                // cglib proxy object
+                aClass = aClass.getSuperclass();
+            }
+            AnnotationMetadata annotationMetadata = AnnotationMetadata.introspect(aClass);
+            AnnotationAttributes attributes = AnnotationAttributes
+                    .fromMap(annotationMetadata
+                            .getAnnotationAttributes(EnableNacos.class.getName()));
+
+            // Register Global Nacos Properties Bean
+            registerGlobalNacosProperties(attributes, registry, environment,
```

Comment:
Why are registerGlobalNacosProperties in the for loop？

Reply:
I made a mistake here, it has been changed to only handle the first bean